tests/unit/test_tools_archive.py

```diff
@@ -0,0 +1,171 @@
+    assert issue_100.number == 100
+    assert issue_100.title == 'tamala2010.example.org - A Punk Cat in Space'
+    assert issue_100.comments_number == 0
+    assert issue_100.comments_url == 'https://api.github.com/repos/webcompat/web-bugs/issues/100/comments'  # noqa
+
+
+def test_issue_archived_header():
+    """Test that the page has an archived notice."""
+    archive_msg = 'This issue has been archived. It is now read-only.'
+    archived_issue = get_fixture('issue_100.html')
+    assert archive_msg in archived_issue
+
+
+def test_issue_links_to_home():
+    """Test that the template has a link to the home page."""
+    archived_issue = get_fixture('issue_100.html')
+    assert '<a href="/" title="Navigate to main page."' in archived_issue
+
+
+def test_save_issue_in_file(tmp_path, issue_100):
+    """Test that we saved the issue in a static folder."""
+    # Create a temporary Path object for testing
+    tmp_root = tmp_path
+    expected_location = tmp_root / 'static' / 'issue' / 'issue-100.html'
+    expected_archived = get_fixture('issue_100.html')
+    # Test it returns the full path to the issue
+    location = issue_100.save(root_dir_path=tmp_root)
+    assert type(location) == pathlib.PosixPath
+    assert expected_location == location
+    # Test it has written the correct content
+    assert expected_archived == location.read_text()
+
+
+def test_issue_has_no_comments_default(issue_100):
+    """Test that an issue has no comments per default."""
+    assert not issue_100.has_comments()
+
+
+def test_issue_has_comments(issue_1470):
+    """Test that an issue has comments."""
+    assert issue_1470.has_comments()
+    assert issue_1470.comments_number == 41
+
+
+def test_comments_fetch(mocker, issue_1470):
+    """Test the comments fetching.
+
+    return a list of dictionary, each dictionary represents a comment.
+    """
+    # Before fetching the comments
+    assert type(issue_1470.comments) == list
+    assert len(issue_1470.comments) == 0
+    # Instead of fetching from the network, we get a fixture
+    fake_json = mocker.patch.object(model.Issue, 'recursive_fetch')
+    fake_json.return_value = json.loads(get_fixture('1470-comments-all.json'))
+    issue_1470.fetch_comments()
+    # actual tests after fetching
+    assert len(issue_1470.comments) == 41
+    assert type(issue_1470.comments) == list
+    assert type(issue_1470.comments[0]) == model.Comment
+    comment_0 = issue_1470.comments[0]
+    assert comment_0.author == 'karlcow'
+    assert 'ok. Thanks a lot' in comment_0.body
+    d = datetime.datetime(2015, 7, 28, 9, 25, 3, tzinfo=datetime.timezone.utc)
+    assert comment_0.created_at == d
+    assert comment_0.updated_at == d
+
+
+def test_comments_fetch_no_comments(issue_100):
+    """Test the comments fetching without comments available."""
+    # Before fetching the comments
+    issue_100.fetch_comments()
+    assert len(issue_100.comments) == 0
+
+
+def test_make_request(mocker):
```

Comment:
Where does `mocker` come from?

Ah,

https://pypi.org/project/pytest-mock/
`This plugin provides a mocker fixture which is a thin-wrapper around the patching API provided by the mock package: Besides undoing the mocking automatically after the end of the test, it also provides other nice utilities such as spy and stub, and uses pytest introspection when comparing calls.`

Seems cool.

Reply:
yup. It's added to the requirements-dev.txt